src/components/EmailAuth.vue

```diff
@@ -1,99 +1,69 @@
-          class="input__field"
-          type="text"
-          :placeholder="$t('login.last-name')"
-          autocomplete="family-name"
-        >
-        <div
-          v-if="!$v.formData.lastname.required"
-          class="error"
-        >
-          {{ $t('form.required-field') }}
-        </div>
-      </div>
-      <div
+        v-model="formData.lastname"
+        :validator="$v.formData.lastname"
+        id="lastname"
+        :label="$t('login.last-name')"
+        :placeholder="$t('login.last-name')"
+        autocomplete="family-name"
+      />
+      <form-input-field
         v-if="!isLinkingProviderPassword"
-        :class="['input', { 'input--error': $v.formData.email.$error }]"
+        v-model.trim="formData.email"
+        :validator="$v.formData.email"
+        id="email"
+        label="Email"
+        placeholder="Email"
+        type="email"
+        :autocomplete="isLinkingProviderPassword ? 'off' : 'email'"
       >
-        <label for="email">
-          Email
-        </label>
-        <input
-          id="email"
-          v-model.trim="$v.formData.email.$model"
-          class="input__field"
-          type="email"
-          placeholder="Email"
-          :autocomplete="isLinkingProviderPassword ? 'off' : 'email'"
-        >
-        <div
-          v-if="!$v.formData.email.required"
-          class="error"
-        >
-          {{ $t('form.required-field') }}
-        </div>
-        <div
-          v-if="!$v.formData.email.email"
-          class="error"
-        >
-          {{ $t('login.invalid-email-address')}}
-        </div>
-      </div>
-      <div :class="['input', { 'input--error': $v.formData.password.$error }]">
-        <label for="password">
-          {{ $t('login.password') }}
-        </label>
-        <input
-          id="password"
-          v-model="$v.formData.password.$model"
-          class="input__field"
-          type="password"
-          :placeholder="$t('login.password')"
-          :autocomplete="login ? 'current-password' : 'new-password'"
-        >
-        <div
-          v-if="!$v.formData.password.required"
-          class="error"
-        >
-          {{ $t('form.required-field') }}
-        </div>
-      </div>
+        <template #validation>
+           <div
+            v-if="!$v.formData.email.required"
+            class="error"
+          >
+            {{ $t('form.required-field') }}
+          </div>
+          <div
+            v-if="!$v.formData.email.email"
+            class="error"
+          >
+            {{ $t('login.invalid-email-address')}}
+          </div>
+        </template>
+      </form-input-field>
+      <form-input-field
+        v-if="!resetPassword"
+        v-model="formData.password"
+        :validator="$v.formData.password"
+        id="password"
+        :label="$t('login.password')"
+        :placeholder="$t('login.password')"
+        type="password"
+        :autocomplete="login ? 'current-password' : 'new-password'"
```

Comment:
do you check to use redirection after updating password and then, set autocomplete to `off`? 
Please check if we can use this method: https://firebase.google.com/docs/auth/web/passing-state-in-email-actions